apps/aurora-portal/src/server/Project/routers/projectRouter.ts

```diff
@@ -20,6 +20,42 @@ export const projectRouter = {
     return parsedData.data.projects
   }),
 
+  searchProjects: protectedProcedure
+    .input(
+      z.object({
+        search: z.string().optional(),
+      })
+    )
+    .query(async ({ ctx, input }): Promise<Project[] | undefined> => {
+      const token = ctx.openstack?.getToken()
+      if (!token) throw new Error("Not authenticated")
```

Comment:
this should be already done using the protectedProcedure

Reply:
what do you mean?